content/blog/2018-06-27-implementing-custom-sd.md

```diff
@@ -0,0 +1,185 @@
+---
+title: Implementing Custom Service Discovery
+created_at: 2018-06-27
+kind: article
+author_name: Callum Styan
+---
+
+## Implementing Custom Service Discovery
+
+Prometheus contains built in integrations for many service discovery (SD) systems such as Consul,
+Kubernetes, and public cloud providers such as Azure. However, we can’t provide integration
+implementations for every service discovery option out there. The Prometheus team is already stretched
+thin supporting the current set of SD integrations, so maintaining an integration for every possible SD
+option isn’t feasible. In many cases the current SD implementations have been contributed by people
+outside the team and then not maintained or tested well. We want to commit to only providing direct
+integration with service discovery mechanisms that we know we can maintain, and that work as intended.
+For this reason, there is currently a moratorium on new SD integrations
+
+However, we know there is still a desire to be able to integrate with other SD mechanisms, such as 
+Docker Swarm. Recently a small code change plus an example was committed to the documentation directory
+within the Prometheus repository for implementing a custom service discovery integration without having
+to merge it into the main Prometheus binary. The code change allows us to make use of the internal
```

Comment:
We should probably make it clear here that we're not going to merge them into the docs directory either, we've had that confusion with the remote adapter example.

We should link to https://prometheus.io/docs/operating/integrations/#file-service-discovery too